server/routes/createReport.ts

```diff
@@ -66,30 +66,15 @@ export default class CreateReport {
         input: req.body,
       })
 
-      const formPayload = { ...payloadFields }
+      const updatedSection = { ...payloadFields }
 
       if (!isNilOrEmpty(errors)) {
         req.flash('errors', errors)
-        req.flash('userInput', formPayload)
+        req.flash('userInput', updatedSection)
         return res.redirect(req.originalUrl)
       }
 
-      const { formId, form } = await this.loadForm(req)
-
-      const updatedPayload = mergeIntoPayload({
-        formObject: form,
-        formPayload,
-        formName,
-      })
-
-      if (updatedPayload) {
-        await this.draftReportService.update({
-          currentUser: res.locals.user,
-          formId,
-          bookingId: parseInt(bookingId, 10),
-          formObject: updatedPayload || {},
-        })
-      }
```

Comment:
pushing this logic down (and also duplicated logic in incident details) into service layer